### PR TITLE
Implementing `ansible-test` for PRs via GitHub Actions

### PR DESCRIPTION
**Pull request by exp-hc**
This change contains a lot of reformatting and refactoring to pass `ansible-test` sanity testing documented [here](https://docs.ansible.com/ansible/latest/dev_guide/testing_sanity.html). The idea is to have `ansible-test` run for ansible 2.10, 2.11, and 2.12 each time a PR is submitted or a push occurs on the `main` branch. This should help the team standardize our code and ensure our documentation matches the arg specs for each module.

Here is a brief list of changes implemented from the [PR](https://github.com/exp-hc/ansible-collection-elastic/pull/1) in my fork:

* Create ansible-test.yml for GitHub Action workflow

* sanity ignore files

* updated formatting per pep8 and pylint

* updated all documentation

* formatted with black and line length increased to 160

* refactored bare excepts

* removed unused imports

* Module Utils `port` arg updated to string

* Updated galaxy.yml to 1.3.0 due to major refactoring

I've created this as a draft until I can get instructions for future contributions and testing locally but I wanted to get this in front of several people for testing existing jobs and updating documentation that contains placeholders. 

There are a lot of changes here so please ask questions or reach out if I need to clarify anything.

### CODE REVIEW

**Comment by exp-hc** [comment]
I should also note, this is currently available to download on Tower-Hub as `1.3.0-dev1` if you want to test this in Tower.

**Comment by exp-hc** [comment]
I've updated the Actions settings for this repo to allow the [ansible-test workflow from ansible-community](https://github.com/ansible-community/ansible-test-gh-action) as an approved workflow.

**Comment by exp-hc** [comment]
@mgaruccio  and @ianrscottexp this PR has been updated to include all the changes in the 1.3.X releases. I'd like to get this merged before we may any further changes in main. Each update is pretty frustrating to merge since this converts all modules from 2 space indentions to 4 spaces, which means git sees all merged changes as conflicting.

File: docs/expedient.elastic.ece_snapshot_repo_module.rst [hunk 1]
```diff
@@ -8,7 +8,7 @@ expedient.elastic.ece_snapshot_repo
 **Create or delete a snapshot repository**
 
 
-Version added: 2.9
+Version added: 1.0.0
```
Comment (mgaruccio) [inline docs/expedient.elastic.ece_snapshot_repo_module.rst:11]: am I understanding "version added" wrong? I thought this was to reference an ansible version?

**Review by mgaruccio — APPROVED** [review]
Super minor comments but I'm good with merging even without edits since none of it is actual code

File: README.md [hunk 2]
```diff
@@ -26,19 +26,45 @@ You can also include it in a `requirements.yml` file and install it via `ansible
 collections:
 - name: expedient.elastic
 ```
+
+### Required Python libraries
+
+VMware community collection depends upon following third party libraries:
+
+* [`PyYAML`](https://pyyaml.org/wiki/PyYAMLDocumentation)
+
```
[the remark this answers is not in the document]
Comment (exp-hc) [reply README.md:35]: that is a requirement in ece_cluster.py but I can fix the "VMware" bit, that was a copy paste error.

File: docs/expedient.elastic.ece_snapshot_repo_module.rst (hunk 1 above)
Comment (mgaruccio) [reply docs/expedient.elastic.ece_snapshot_repo_module.rst:11]: ok yea I misunderstood that field, thanks for clarifying

**Comment by mgaruccio** [comment]
@exp-hc  - are you still good to merge this?  want to get it done before we do Ian's next one so we don't need to keep doing conflict resolves

**Comment by ianrscottexp** [comment]
Just made a minor update based on Hunter's comment.  I am good to PR if you guys are.

**Comment by exp-hc** [comment]
@mgaruccio I need to catch-up to the last commit so we should go ahead with Ian's PR for now. I'll get this caught up before the end of the month.

**Pull request closed without merging**